microsoft-edge/webview2/concepts/process-related-events.md

```diff
@@ -0,0 +1,293 @@
+* [ICoreWebView2Environment5::remove_BrowserProcessExited event](https://learn.microsoft.com/microsoft-edge/webview2/reference/win32/icorewebview2environment5#remove_browserprocessexited)
+
+---
+
+See also:
+* [Process model for WebView2 apps](./process-model.md)
+
+
+<!-- ====================================================================== -->
+## Gathering process failure details
+
+The `ProcessFailed` event provides detailed information about the process failure being reported.  Your application can use and collect information from the event args for monitoring and diagnostics purposes, including process description (for utility processes only) and frames information (for renderer processes only).
+
+##### [.NET/C#](#tab/dotnetcsharp)
+
+* [CoreWebView2ProcessFailedEventArgs Class](https://learn.microsoft.com/dotnet/api/microsoft.web.webview2.core.corewebview2processfailedeventargs)
+
+##### [WinRT/C#](#tab/winrtcsharp)
+
+* [CoreWebView2ProcessFailedEventArgs Class](https://learn.microsoft.com/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2processfailedeventargs)
+
+##### [Win32/C++](#tab/win32cpp)
+
+* [ICoreWebView2ProcessFailedEventArgs2 interface](https://learn.microsoft.com/microsoft-edge/webview2/reference/win32/icorewebview2processfailedeventargs2)
+
+---
+
+Some process failures might raise the `ProcessFailed` event across different WebView2 controls in your application.  You must decide how often to gather details and how to handle duplicates for these cases.
+
+Additionally, most process crashes will generate dumps in the [user data folder](./user-data-folder.md), under the directory returned by `FailureReportFolderPath`. You can use these dumps to understand crashes and provide additional information when contacting the WebView2 team.
+
+##### [.NET/C#](#tab/dotnetcsharp)
+
+* [CoreWebView2Environment.FailureReportFolderPath Property](https://learn.microsoft.com/dotnet/api/microsoft.web.webview2.core.corewebview2environment.failurereportfolderpath)
+
+##### [WinRT/C#](#tab/winrtcsharp)
+
+* [CoreWebView2Environment.FailureReportFolderPath Property](https://learn.microsoft.com/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2environment#failurereportfolderpath)
+
+##### [Win32/C++](#tab/win32cpp)
+
+* [ICoreWebView2Environment11::get_FailureReportFolderPath](https://learn.microsoft.com/microsoft-edge/webview2/reference/win32/icorewebview2environment11#get_failurereportfolderpath)
+
+---
+
+
+<!-- ====================================================================== -->
+## Handle process crashes
+
+When a crash occurs in the WebView2 Runtime, the `ProcessFailed` event will be raised for every WebView2 control associated to the crashing process.  The failure might or might not be recoverable, and some failures are auto-recoverable.
+
+You can use the following event args to identify the failure:
```

Comment:
These are properties from the event args. Should we link back to 'Gathering' section which has the link to `ProcessFailedEventArgs` or otherwise link to API reference for `ProcessFailedKind` and `ProcessFailedReason` enums?

@aluhrs13 thoughts?

Reply:
FWIW I think we should add the tab to link to the specific properties like we do throughout the section.

Reply:
"link to the specific properties" - which?
Checking: are Kind & Reason properties of eventArgs? __

Reply:
locally, added:
```
##### [.NET/C#](#tab/dotnetcsharp)

* [CoreWebView2ProcessFailedEventArgs Class](https://learn.microsoft.com/dotnet/api/microsoft.web.webview2.core.corewebview2processfailedeventargs)
* [CoreWebView2ProcessFailedKind Enum](/dotnet/api/microsoft.web.webview2.core.corewebview2processfailedkind)
* [CoreWebView2ProcessFailedReason Enum](/dotnet/api/microsoft.web.webview2.core.corewebview2processfailedreason)

##### [WinRT/C#](#tab/winrtcsharp)

* [CoreWebView2ProcessFailedEventArgs Class](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2processfailedeventargs)
* [CoreWebView2ProcessFailedKind Enum](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2processfailedkind)
* [CoreWebView2ProcessFailedReason Enum](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2processfailedreason)

##### [Win32/C++](#tab/win32cpp)

* `ICoreWebView2ProcessFailedEventArgs` interface:
   * [ICoreWebView2ProcessFailedEventArgs::get_ProcessFailedKind property](https://learn.microsoft.com/microsoft-edge/webview2/reference/win32/icorewebview2processfailedeventargs#get_processfailedkind)
* `ICoreWebView2ProcessFailedEventArgs2` interface:
   * [ICoreWebView2ProcessFailedEventArgs2::get_Reason property](https://learn.microsoft.com/microsoft-edge/webview2/reference/win32/icorewebview2processfailedeventargs2#get_reason)
* [COREWEBVIEW2_PROCESS_FAILED_KIND enum](/microsoft-edge/webview2/reference/win32/webview2-idl#corewebview2_process_failed_kind)
* [COREWEBVIEW2_PROCESS_FAILED_REASON enum](/microsoft-edge/webview2/reference/win32/webview2-idl#corewebview2_process_failed_reason)

---

See also:
* [Gathering process failure details](#gathering-process-failure-details)
```